linter.py

```diff
@@ -120,6 +138,15 @@ def ensure_plugin_installed(self) -> bool:
         self.notify_unassign()  # Abort linting without popping error dialog
         raise PermanentError()
 
+    def get_stdin_filename(self) -> str:
+        filename = self.view.file_name()
+        if filename is None:
+            for selector in BUFFER_FILE_EXTENSIONS.keys():
+                if self.view.match_selector(0, selector):
+                    filename = '.'.join([BUFFER_FILE_STEM, BUFFER_FILE_EXTENSIONS[selector]])
+                    break
+        return filename
```

Comment:
This should work but `self.view.match_selector` is a side-effect that requires locking (or syncing) between python and Sublime core *each* time.  

I *think* `self.view.scope_name(0)` (https://www.sublimetext.com/docs/api_reference.html#sublime.View.scope_name) should be used here.   For example

```
selectors = set(self.view.scope_name(0).split(' '))
```
once will probably allow a simple `if selector in selectors` in the `for` loop.  (Don't know if `selectors` is the correct variable name here because I actually don't know the correct nomenclature here, t.i. what is a selector, what is a scope_name?) 🤷